compiler/luci/import/src/Importer.test.cpp

```diff
@@ -16,13 +16,289 @@
+    entry_data_ptr[2] = execution_order;            // entry size
+    std::copy(offsets.begin(), offsets.end(), entry_data_ptr + 3);
+  }
+};
+
+struct SimpleRELUModel : public BasicCircleModel
+{
+  SimpleRELUModel()
+  {
+    auto relu_opcode_id = add_builtin_opcode(circle::BuiltinOperator_RELU);
+
+    uint32_t subgraph_id = add_subgraph();
+
+    auto input_buffer_id = add_buffer();
+    auto output_buffer_id = add_buffer();
+
+    auto input_tensor_idx = add_float_tensor(subgraph_id, {1, 2, 3, 4}, input_buffer_id);
+    auto output_tensor_idx = add_float_tensor(subgraph_id, {1, 2, 3, 4}, output_buffer_id);
+
+    add_subgraph_inputs(subgraph_id, {input_tensor_idx});
+    add_subgraph_outputs(subgraph_id, {output_tensor_idx});
+
+    add_builtin_operator(subgraph_id, relu_opcode_id, {0}, {1});
+  }
+};
+
+} // namespace
+
+/**
+ * This test checks that one op RELU model with execution plan is successfully imported
+ */
+TEST(TensorFlowLiteImporter, simple_plan)
+{
+  SimpleRELUModel model;
+  auto metadata_buffer_id = model.add_buffer();
+  model.add_plan_metadata(metadata_buffer_id);
+
+  model.add_plan_entry(metadata_buffer_id, 1, {100});
+  model.add_plan_entry(metadata_buffer_id, 2, {300});
+  model.add_plan_entry(metadata_buffer_id, 3, {200});
+
+  flatbuffers::FlatBufferBuilder fbb;
+  auto model_offset = circle::Model::Pack(fbb, model.model.get(), nullptr);
+  circle::FinishModelBuffer(fbb, model_offset);
+
+  auto model_ptr = circle::GetModel(fbb.GetBufferPointer());
+  luci::Importer import;
+
+  auto luci_module = import.importModule(model_ptr);
+
+  auto main_graph = luci_module->graph();
+  for (int i = 0; i < main_graph->nodes()->size(); ++i)
+  {
+    auto node = loco::must_cast<luci::CircleNode *>(main_graph->nodes()->at(i));
+    switch (node->opcode())
+    {
+      case luci::CircleOpcode::CIRCLEINPUT:
+      {
+        ASSERT_TRUE(luci::has_execution_plan(node));
+        auto plan = luci::get_execution_plan(node);
+        ASSERT_EQ(plan.order_in_plan(), 1);
+        ASSERT_EQ(plan.offsets().size(), 1);
+        ASSERT_EQ(plan.offsets()[0], 100);
+        break;
+      }
+      case luci::CircleOpcode::CIRCLEOUTPUT:
+      {
+        ASSERT_TRUE(luci::has_execution_plan(node));
+        auto plan = luci::get_execution_plan(node);
+        ASSERT_EQ(plan.order_in_plan(), 3);
+        ASSERT_EQ(plan.offsets().size(), 1);
+        ASSERT_EQ(plan.offsets()[0], 200);
+        break;
+      }
+      case luci::CircleOpcode::RELU:
+      {
+        ASSERT_TRUE(luci::has_execution_plan(node));
+        auto plan = luci::get_execution_plan(node);
+        ASSERT_EQ(plan.order_in_plan(), 2);
+        ASSERT_EQ(plan.offsets().size(), 1);
+        ASSERT_EQ(plan.offsets()[0], 300);
+        break;
+      }
+    }
+  }
+}
+
+/**
+ * This test checks that model with incomplete execution plan is successfully imported
+ */
+TEST(TensorFlowLiteImporter, DISABLED_incomplete_plan_NEG)
```

Comment:
This test is questionable.
The problem with it is following:
- Exporter theoretically can produce models with incomplete execution plan (no data for some operations)
- Importer throws on attempt to import such models.

From my point of view this is inconsistency of interfaces, we need to decide, should we:
1. Throw in exporter in case of incomplete plan?
2. Successfully import models with some data missing?

Reply:
> Throw in exporter in case of incomplete plan?

I disagree on this. As I understand `plan` is optional, and I don't know how to use this `plan`

Reply:
> I disagree on this. As I understand plan is optional, and I don't know how to use this plan

You are right, it is optional.

There are three cases:
- All nodes has plan metadata
- All nodes does not have plan metadata
- Some nodes has, some does not

First two cases are fine, third one is inconsistent:
- Exporter can produce such circle model
- Importer can not parse it =(

I just thought that there are third solution:
- discard all plan annotation in case some of the them are missing

Reply:
I think serialize plan partially (2 case from original post) is the best solution.

- Exception case. After applying planning tool we can not do any transformations. Otherwise we need to explicitly remove plan annotations. It is not clear for me who should do this cleaning (this is also hard to describe to user :slightly_smiling_face: )
- Implicit discarding. This is also hard do describe and IMHO a little confusing

Reply:
Lets merge this PR, then fix execution plan importing and enable this test.
Related issue #7907